src/mid.cpp

```diff
@@ -300,7 +300,8 @@ bool CmidPlayer::load(const std::string &filename, const CFileProvider &fp)
     uint32_t size;
 
     f->readString((char *)s, 6);
-    size = *(uint32_t *)s; // size of FILE_OLDLUCAS
+    // need to perform unaligned access
+    memcpy(&size, s, sizeof(uint32_t)); // size of FILE_OLDLUCAS
```

Comment:
This will still yield different results on big endian vs little.  Might be good to make it endian neutral at the same time.

Reply:
Correct, I wasn't going to fix the endianness issues with this PR yet

Reply:
That's fine, but I suspect the endian issues are going to be significant enough that it may involve a second rewrite, so might be a good idea to try to fit both in at the same time if possible, just to save some effort.